sympy/assumptions/tests/test_satask.py

```diff
@@ -0,0 +1,254 @@
+from sympy.assumptions.satask import satask
+
+from sympy import symbols, Q, assuming, Implies, MatrixSymbol, I, pi, Rational
+
+from sympy.utilities.pytest import raises, XFAIL
+
+
+x, y, z = symbols('x y z')
+
+
+def test_satask():
+    # No relevant facts
+    assert satask(Q.real(x), Q.real(x)) is True
+    assert satask(Q.real(x), ~Q.real(x)) is False
+    assert satask(Q.real(x)) is None
+
+    assert satask(Q.real(x), Q.positive(x)) is True
+    assert satask(Q.positive(x), Q.real(x)) is None
+    assert satask(Q.real(x), ~Q.positive(x)) is None
+    assert satask(Q.positive(x), ~Q.real(x)) is False
+
+    raises(ValueError, lambda: satask(Q.real(x), Q.real(x) & ~Q.real(x)))
+
+    with assuming(Q.positive(x)):
+        assert satask(Q.real(x)) is True
+        assert satask(~Q.positive(x)) is False
+        raises(ValueError, lambda: satask(Q.real(x), ~Q.positive(x)))
+
+    assert satask(Q.zero(x), Q.nonzero(x)) is False
+    assert satask(Q.positive(x), Q.zero(x)) is False
+    assert satask(Q.real(x), Q.zero(x)) is True
+    assert satask(Q.zero(x), Q.zero(x*y)) is None
+    assert satask(Q.zero(x*y), Q.zero(x))
+
+
+def test_zero():
+    """
+    Everything in this test doesn't work with ask, and most things would be
+    very difficult or impossible to make work under the current handlers
+    model.
+    """
```

Comment:
That looks good. I'm okay with the merge now.

On 6 Jun 2015, at 19:34, Aaron Meurer wrote:

>> +        assert satask(~Q.positive(x)) is False
>> +        raises(ValueError, lambda: satask(Q.real(x), 
>> ~Q.positive(x)))
>> +
>> +    assert satask(Q.zero(x), Q.nonzero(x)) is False
>> +    assert satask(Q.positive(x), Q.zero(x)) is False
>> +    assert satask(Q.real(x), Q.zero(x)) is True
>> +    assert satask(Q.zero(x), Q.zero(x*y)) is None
>> +    assert satask(Q.zero(x*y), Q.zero(x))
>> +
>> +
>> +def test_zero():
>> +    """
>> +    Everything in this test doesn't work with ask, and most things 
>> would be
>> +    very difficult or impossible to make work under the current 
>> handlers
>> +    model.
>> +    """
>
> I modified it.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/sympy/sympy/pull/2508/files#r31872394